delfin/exporter/prometheus/prometheus.py

```diff
@@ -41,26 +50,47 @@
      values={1600998817585: 20.264160223426305})]
 """
 
-unit_of_metric = {'response_time': 'ms', 'throughput': 'IOPS',
-                  'read_throughput': 'IOPS', 'write_throughput': 'IOPS',
-                  'bandwidth': 'MBps', 'read_bandwidth': 'MBps',
-                  'write_bandwidth': 'MBps'
-                  }
-
 
 class PrometheusExporter(object):
 
+    def __init__(self):
+        self.timestamp_offset_ms = self.set_timestamp_offset_from_utc_ms()
+        self.metrics_dir = cfg.CONF.PROMETHEUS_EXPORTER.metrics_dir
+
+    def set_timestamp_offset_from_utc_ms(self):
+        """Set timestamp offset from utc required for all metrics"""
+        try:
+            timez = get_localzone()
+            if cfg.CONF.PROMETHEUS_EXPORTER.timezone != 'local':
+                timez = pytz.timezone(cfg.CONF.PROMETHEUS_EXPORTER.timezone)
+            timez.utcoffset(datetime.datetime.now())
+            return int(timez.utcoffset(
+                datetime.datetime.now()).total_seconds() * 1000)
+        except Exception:
+            LOG.error('Error while setting timestamp'
+                      ' offset for prometheus exporter')
+            # return no offset in case of an error
+            return 0
+
     # Print metrics in Prometheus format.
-    def _write_to_prometheus_format(self, f, metric, labels, values):
-        f.write("# HELP storage_%s storage metric for %s\n" % (metric, metric))
-        f.write("# TYPE storage_%s gauge\n" % metric)
+    def _write_to_prometheus_format(self, f, metric,
+                                    labels, prom_labels, values):
+        f.write("# HELP %s  metric for resource %s and instance %s\n"
+                % (metric, labels.get('resource_type'),
+                   labels.get('resource_id')))
+        f.write("# TYPE %s gauge\n" % metric)
 
         for timestamp, value in values.items():
-            f.write("storage_%s{%s} %f %d\n" % (metric, labels,
-                                                value, timestamp))
+            timestamp += self.timestamp_offset_ms
```

Comment:
Timestamp update can be moved out of _write_to_prometheus_format() function.

Reply:
values dictionary contains multiple timestamps.. each timestamp is distinct .. so adding offset has to be inside this loop